tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1353,11 +1353,26 @@ static shared_ptr<DuckDBPyConnection> FetchOrCreateInstance(const string &databa
 	return res;
 }
 
+bool IsDefaultConnectionString(const string &database, bool read_only, unordered_map<string, string> &config) {
+	bool is_default = StringUtil::CIEquals(database, ":default:");
+	if (!is_default) {
+		return false;
+	}
+	// Only allow fetching the default connection when no options are passed
+	if (read_only == true || !config.empty()) {
```

Comment:
Should this not be `&&`?

Reply:
The default value for `read_only` is true, and the config is empty by default.
If either of them are not default, then I think we should throw, since we're just retrieving an existing connection.
I figured if options are provided and we just silently ignore them that could be confusing to the user

Reply:
That does mean that `con = duckdb.connect(':memory:', read_only=False)` does work though.
I could change the `read_only` parameter to an object, making it None by default so we can properly check if it's not provided

Reply:
Ah yes, that makes sense. I think it is fine the way it is.